### ISSUE TITLE
Galilean CI tests

### ISSUE THREAD

**Issue by EZoni** [ECP-WarpX/WarpX#1163]
When I try to run the test **galilean_2d_psatd** with zero Galilean velocity, setting `psatd.v_galilean = 0. 0. 0.` in Examples/Tests/galilean/inputs_2d, the resulting electric energy seems to be
```py
energyE_psatd = 33286.43069686019
```
instead of 
https://github.com/ECP-WarpX/WarpX/blob/8af11ca6108147464c4072ae62e643582bc15563/Examples/Tests/galilean/analysis_2d.py#L28-L29

<br/>

Similarly, when I try to run the test **galilean_3d_psatd** with zero Galilean velocity, setting `psatd.v_galilean = 0. 0. 0.` in Examples/Tests/galilean/inputs_3d, the resulting electric energy seems to be
```py
energyE_psatd = 403.7990470629009
```
instead of
https://github.com/ECP-WarpX/WarpX/blob/8af11ca6108147464c4072ae62e643582bc15563/Examples/Tests/galilean/analysis_3d.py#L28-L29

<br/>

The larger difference here is due to the fact that this test is currently 2D and not 3D:
https://github.com/ECP-WarpX/WarpX/blob/8af11ca6108147464c4072ae62e643582bc15563/Regression/WarpX-tests.ini#L1428-L1431

<br/>

When I try to run the same test in 3D, by setting `dim = 3` above, the resulting electric energy seems to be
```py
energyE_psatd = 8946.257477449579
```
still smaller than the energy reference set in Examples/Tests/galilean/analysis_3d.py.

<br/>

Also, I am not sure that we are reading the field diagnostics correctly in the checksum regression analysis for these two tests, as I see that all fields seem to be zero, which I think is not compatible with the energy values above:
https://github.com/ECP-WarpX/WarpX/blob/8af11ca6108147464c4072ae62e643582bc15563/Regression/Checksum/benchmarks_json/galilean_2d_psatd.json#L22-L32
https://github.com/ECP-WarpX/WarpX/blob/8af11ca6108147464c4072ae62e643582bc15563/Regression/Checksum/benchmarks_json/galilean_3d_psatd.json#L22-L32

<br/>

I discussed some of this already offline with @oshapoval, but, since I made a few more tests and observations, I thought it would be easier to collect the information here. I might be interpreting the reference energy values in the wrong way, but I think it might be worth cross-checking all these numbers. Especially if the correct energy references turn out to be smaller than the one currently used in the analysis, which would make the corresponding tests stronger (since we are measuring instabilities).

**Comment by oshapoval** [comment]
@EZoni Thanks a lot for sharing your observations!
I recalculated the reference energy values for both 2D and 3D tests with v_galilean = (0,0,0) using:` ./run_test.sh galilean_2d_psatd galilean_3d_psatd`.
I am confirming that I am getting same energy as you did for 2D case (`energyE_psatd = 403.7990470629009`) and slightly different for 3D case (`energyE_psatd = 414.22472100727856` and `energyE_psatd = 7725.714751093252`, respectively). I will change wrong values, thanks. My guess is that I was using another WarpX’s executable for energy calculation (with compilation options DEBUG=TRUE and/or USE_CCACHE=TRUE) instead of the `main2d.gnu.TEST.TPROF.MPI.OMP.PSATD.ex` or `main3d.gnu.TEST.TPROF.MPI.OMP.PSATD.ex` or even calculated for other time step.

**Comment by oshapoval** [comment]
Sorry, previous comment was not complete.

**Comment by oshapoval** [comment]
@EZoni Sorry, previous comment was incomplete. There is a typo in the `WarpX/Regression/WarpX-tests.ini` from my side. It should be `dim=3` instead of `dim=2` for `galilean_3d_psatd` test. I noticed it when was adding 3D automated test for the averaged algorithm. I will update it.

**Comment by EZoni** [comment]
@oshapoval Thank you very much for checking this!

Regarding the checksum benchmarks, the reason for having all zeros seems to be how the fields data are read in [Regression/Checksum/checksum.py](https://github.com/ECP-WarpX/WarpX/blob/development/Regression/Checksum/checksum.py).

In your analysis scripts, the fields data are read with
https://github.com/ECP-WarpX/WarpX/blob/0b00cfa973de24e3ae08ecd0894b2d9b8b986c0d/Examples/Tests/galilean/analysis_2d.py#L19-L23
which seems to give the correct values (given that the corresponding energy is correct).

<br/>

On the contrary, in Regression/Checksum/checksum.py the fields data are read with
https://github.com/ECP-WarpX/WarpX/blob/0b00cfa973de24e3ae08ecd0894b2d9b8b986c0d/Regression/Checksum/checksum.py#L52-L63
which seems to give the wrong results, namely all zeros, as suggested by the JSON files mentioned in the description of the issue.

<br/>

@MaxThevenet @atmyers 
Do you maybe have an idea of why reading the field data as it is done in Regression/Checksum/checksum.py does not work correctly for these particular tests?

**Comment by EZoni** [comment]
Most parts of the issue were fixed by @oshapoval's PR #1173!

Only the 2D JSON benchmarks for the Galilean and averaged Galilean tests remain to be fixed, by either hacking Regression/Checksum/checksum.py for these specific tests or by fixing the corresponding issue on the **yt** side (if any):

https://github.com/ECP-WarpX/WarpX/blob/af7144301bf0d6bbc2263ef389c321cb0cb6fe40/Regression/Checksum/benchmarks_json/galilean_2d_psatd.json#L22-L32

https://github.com/ECP-WarpX/WarpX/blob/af7144301bf0d6bbc2263ef389c321cb0cb6fe40/Regression/Checksum/benchmarks_json/averaged_galilean_2d_psatd.json#L22-L32
